### PR TITLE
bump git2 dependencies

### PR DESCRIPTION
**Pull request by matthiaskrgr**
This required some manual code changes which dependabot could not perform.

### CODE REVIEW

**Comment by Eh2406** [comment]
Ci seems unhappy.

File: crates/cargo-test-support/src/git.rs [hunk 1]
```diff
@@ -197,7 +197,7 @@ pub fn add_submodule<'a>(
     default_repo_cfg(&subrepo);
     t!(subrepo.remote_add_fetch("origin", "refs/heads/*:refs/heads/*"));
     let mut origin = t!(subrepo.find_remote("origin"));
-    t!(origin.fetch(&[], None, None));
+    t!(origin.fetch(&[&String::new()], None, None));
```
Comment (ehuss) [inline crates/cargo-test-support/src/git.rs]: Can you explain why this change is needed?  The [docs](https://github.com/libgit2/libgit2/blob/2f6f10bb804becddfae78bb9682183ae8ae8184d/include/git2/remote.h#L819-L820) still say an empty array is OK.

Comment (ehuss) [reply crates/cargo-test-support/src/git.rs]: Oh, I see, it is due to https://github.com/rust-lang/git2-rs/pull/482.  Perhaps it could use a cast instead?

Comment (alexcrichton) [reply crates/cargo-test-support/src/git.rs]: Yes I believe this is the cause of the broken CI as well, and agree that this should use an explicit cast and/or type parameter to convince the build it should work.

**Comment by Eh2406** [comment]
Ci is green so:
@bors r+
Thanks for the PR!

File: tests/testsuite/git.rs [hunk 2]
```diff
@@ -1201,7 +1201,7 @@ fn dep_with_changed_submodule() {
             .remote_set_url("origin", &git_project3.url().to_string())
             .unwrap();
         let mut origin = subrepo.find_remote("origin").unwrap();
-        origin.fetch(&[], None, None).unwrap();
+        origin.fetch(&Vec::<String>::new(), None, None).unwrap();
```
Comment (lzutao) [inline tests/testsuite/git.rs]:
```suggestion
        origin.fetch(&[] as &[&str], None, None).unwrap();
```

File: crates/cargo-test-support/src/git.rs [hunk 3]
```diff
@@ -197,7 +197,7 @@ pub fn add_submodule<'a>(
     default_repo_cfg(&subrepo);
     t!(subrepo.remote_add_fetch("origin", "refs/heads/*:refs/heads/*"));
     let mut origin = t!(subrepo.find_remote("origin"));
-    t!(origin.fetch(&[], None, None));
+    t!(origin.fetch(&Vec::<String>::new(), None, None));
```
Comment (lzutao) [inline crates/cargo-test-support/src/git.rs]:
```suggestion
    t!(origin.fetch(&[] as &[&str], None, None));
```
